### DISCUSSION TITLE
Get "We're Sorry, but Something Went Wrong" when navigating to even models tab.

### DISCUSSION THREAD

**Discussion by hagenman** [manyfold3d/manyfold#3277] [Support]
**Describe the bug**
After running Manyfold-Solo for a bit, I navigate anywhere and get the "We're Sorry, but Something Went Wrong" page.

**Additional context**
Adding Logs:
[hagen-manyfold-app-1_logs.txt](https://github.com/user-attachments/files/18011813/hagen-manyfold-app-1_logs.txt)

**Comment by hagenman** [comment]
I've spent a lot of time trying to figure this out before opening the bug request, but I'm stumped.

**Reply by Floppy** [reply]
There's a few strange things in the log, but the one causing the error at the end is an unknown postgres adapter, which is very odd because that should be there (even if you're not using postgres).

Have you had manyfold-solo working previously, and have you recently upgraded? What version are you running?

**Reply by hagenman** [reply]
yeah, it was working fine for a while and then just quit.  I'm on version[0.90.1 (7cf85236)] but would be willing to move to the full deployment, I just don't know how to migrate the database.

**Reply by Floppy** [reply]
That's really odd; can I get you to upgrade to 0.91.0 just in case, then make sure none of the host disks are full, the network drives are accessible etc? Can you also post your compose or other config, or docker inspect output?

**Reply by hagenman** [reply]
It's running on my Synology NAS, using Portainer.  Oddly, I had the image tagged to latest, and even after a new pull it was on the old version.  I had to modify the compose file to specidy 0.91.0 tag to get it to upgrade,  

Compose is as follows:
services:
  app:
    image: ghcr.io/manyfold3d/manyfold-solo:0.91.0
    ports:
      - 3214:3214
    volumes:
      - /volume1/docker/manyfold/config:/config
      - /volume1/docker/manyfold/libraries:/libraries
    environment:
      SECRET_KEY_BASE: a_nice_long_random_string
      PUID: 1026
      PGID: 100
    restart: unless-stopped
    # Optional, but recommended for better security
    security_opt:
      - no-new-privileges:true
    cap_drop:
      - ALL
    cap_add:
      - CHOWN
      - DAC_OVERRIDE
      - SETUID
      - SETGID

Logs after upgrade attached.  I clicked Models on the homepage after the upgrade and it immediately went to the error page again.
[hagen-manyfold-app-1_logs(1).txt](https://github.com/user-attachments/files/18025725/hagen-manyfold-app-1_logs.1.txt)

**Reply by swichers** [reply]
Your SQLite database may be corrupt. I also had this issue where everything was working and then suddenly all I could get was the 'sorry' error despite restarting, updating, whatever.

```sh
~ ❯❯❯ sqlite3 manyfold.sqlite3 'PRAGMA integrity_check'
Error: stepping, database disk image is malformed (11)
```

For testing, I removed the existing sqlite database and let the application create a new one. It fired right up without any problems.

The next thing I did was attempt to recover the data:

```sh
~ ❯❯❯ sqlite3 manyfold.sqlite3 .dump > dump.sqlite3
~ ❯❯❯ sqlite3 recovered.sqlite3 '.read dump.sqlite3'
```

(Note: sqlite has a .recover command, but this wasn't working for me)

After that, I backed up my existing sqlite database and put the recovered.sqlite3 (renamed to manyfold) in its place.

This let me actually open the application.

It's worth noting my recovered (bottom) file is significantly smaller than the existing one, but I am not sure what I have lost (if anything) at this time.

![image](https://github.com/user-attachments/assets/4e01c44e-4cd1-4bf9-aa50-526afa78b8f4)

I second wanting information on how to migrate to a different database. Do I just need to import the data from sqlite into mysql, or are there differences in how the data is stored?

**Reply by hagenman** [reply]
Thank you!  PRAGMA integrity_check comes back ok, and I tried the recovery and swap but that made no difference, clicking models brought me to the error page, even though the file did come back smaller like you observed.

**Reply by swichers** [reply]
Bummer. I was hopeful this would be useful since we both had the same symptoms. Good luck.

**Comment by Floppy** [comment]
Thanks to your logs, I found a bug with the database type checks, and have pushed a fix in v0.91.1 - hopefully that should fix this for you!

**Comment by Floppy** [comment]
Your logs also hinted at another issue with analysis jobs, which is now fixed in v0.91.3.
